bundles/org.openhab.binding.wled/src/main/java/org/openhab/binding/wled/internal/WLedHandler.java

```diff
@@ -0,0 +1,456 @@
+import java.math.RoundingMode;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpMethod;
+import org.openhab.core.library.types.HSBType;
+import org.openhab.core.library.types.IncreaseDecreaseType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.PercentType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.StateOption;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link WLedHandler} is responsible for handling commands and states, which are
+ * sent to one of the channels or http replies back.
+ *
+ * @author Matthew Skinner - Initial contribution
+ */
+
+@NonNullByDefault
+public class WLedHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final HttpClient httpClient;
+    private final WledDynamicStateDescriptionProvider stateDescriptionProvider;
+    private @Nullable ScheduledFuture<?> pollingFuture = null;
+    private ScheduledExecutorService threadPool = Executors.newScheduledThreadPool(1);
+    private BigDecimal masterBrightness = new BigDecimal(0);
+    private HSBType primaryColor = new HSBType();
+    private BigDecimal primaryWhite = new BigDecimal(0);
+    private HSBType secondaryColor = new HSBType();
+    private BigDecimal secondaryWhite = new BigDecimal(0);
+    private boolean hasWhite = false;
+    private WLedConfiguration config;
+
+    public WLedHandler(Thing thing, HttpClient httpClient,
+            WledDynamicStateDescriptionProvider stateDescriptionProvider) {
+        super(thing);
+        this.httpClient = httpClient;
+        this.stateDescriptionProvider = stateDescriptionProvider;
+        config = getConfigAs(WLedConfiguration.class);
+    }
+
+    private void sendGetRequest(String url) {
+        Request request = httpClient.newRequest(config.address + url);
+        request.timeout(3, TimeUnit.SECONDS);
```

Comment:
I am happy to use either 10 or 3. I just like 3 seconds as it is more than enough for a device on a local network and you have a faster response to the device going OFFLINE if the power switch is turned off. Otherwise it could be 20 seconds after you turn the light off at the wall that openHAB knows it is offline due to the default 10 second poll and then another 10 for the timeout. If it was talking to remote servers on the web, then I would want 10 seconds.

Reply:
I'm fine with 3 sec. Your arguments are valid.